drivers/sqljs-driver/src/jsTest/kotlin/com/squareup/sqldelight/drivers/sqljs/JsDriverTest.kt

```diff
@@ -0,0 +1,222 @@
+            driver.execute(2, "INSERT INTO test VALUES (?, ?);", 2, binders)
+        }
+        val query = {
+            driver.executeQuery(3, "SELECT * FROM test", 0)
+        }
+        val changes = {
+            driver.executeQuery(4, "SELECT changes()", 0)
+        }
+
+        query().use {
+            assertFalse(it.next())
+        }
+
+        insert {
+            bindLong(1, 1)
+            bindString(2, "Alec")
+        }
+
+        query().use {
+            assertTrue(it.next())
+            assertFalse(it.next())
+        }
+
+        assertEquals(1, changes().apply { next() }.use { it.getLong(0) })
+
+        query().use {
+            assertTrue(it.next())
+            assertEquals(1, it.getLong(0))
+            assertEquals("Alec", it.getString(1))
+        }
+
+        insert {
+            bindLong(1, 2)
+            bindString(2, "Jake")
+        }
+        assertEquals(1, changes().apply { next() }.use { it.getLong(0) })
+
+        query().use {
+            assertTrue(it.next())
+            assertEquals(1, it.getLong(0))
+            assertEquals("Alec", it.getString(1))
+            assertTrue(it.next())
+            assertEquals(2, it.getLong(0))
+            assertEquals("Jake", it.getString(1))
+        }
+
+        driver.execute(5, "DELETE FROM test", 0)
+        assertEquals(2, changes().apply { next() }.use { it.getLong(0) })
+
+        query().use {
+            assertFalse(it.next())
+        }
+
+        tearDown()
+    }
+
+    @JsName("query_can_run_multiple_times")
+    @Test fun `query can run multiple times`() = runTest {
+        setup()
+
+        val insert = { binders: SqlPreparedStatement.() -> Unit ->
+            driver.execute(2, "INSERT INTO test VALUES (?, ?);", 2, binders)
+        }
+        val changes = {
+            driver.executeQuery(4, "SELECT changes()", 0)
+        }
+
+        insert {
+            bindLong(1, 1)
+            bindString(2, "Alec")
+        }
+        assertEquals(1, changes().apply { next() }.use { it.getLong(0) })
+        insert {
+            bindLong(1, 2)
+            bindString(2, "Jake")
+        }
+        assertEquals(1, changes().apply { next() }.use { it.getLong(0) })
+
+        val query = { binders: SqlPreparedStatement.() -> Unit ->
+            driver.executeQuery(6, "SELECT * FROM test WHERE value = ?", 1, binders)
+        }
+        query {
+            bindString(1, "Jake")
+        }.use {
+            assertTrue(it.next())
+            assertEquals(2, it.getLong(0))
+            assertEquals("Jake", it.getString(1))
+        }
+
+        // Second time running the query is fine
+        query {
+            bindString(1, "Jake")
+        }.use {
+            assertTrue(it.next())
+            assertEquals(2, it.getLong(0))
+            assertEquals("Jake", it.getString(1))
+        }
+
+        tearDown()
+    }
+
+    @JsName("SqlResultSet_getters_return_null_if_the_column_values_are_NULL")
```

Comment:
You can just name the tests like this directly. They only run on JS, so there's no point duplicating them.

Reply:
Fixed